File: app/src/main/java/com/dinodevs/greatfitwatchface/theme/ITheme.java
package com.dinodevs.greatfitwatchface.theme;

import com.dinodevs.greatfitwatchface.theme.bin.Background;
import com.dinodevs.greatfitwatchface.theme.bin.Calories;
import com.dinodevs.greatfitwatchface.theme.bin.Scale;
import com.dinodevs.greatfitwatchface.theme.bin.Text;
import com.dinodevs.greatfitwatchface.theme.bin.Time;

public interface ITheme {
    Background getBackground();

    String[] getBattery();

    Text getBatterySpec();

    Scale getBatteryScale();

    String getTimeHand() throws Exception;

    Time getTime();

    String getAmPm(int amPm);

    String getImagePath(int imageIndex);

    Calories getCalories();


//    static ITheme Load(String path) {
//    };
}

File: app/src/main/java/com/dinodevs/greatfitwatchface/theme/ITheme.kt
package com.dinodevs.greatfitwatchface.theme

import com.dinodevs.greatfitwatchface.theme.bin.*

interface ITheme {
    val background: Background
    val battery: Array<String>
    val batterySpec: Text?
    val batteryScale: Scale?
    @get:Throws(Exception::class)
    val timeHand: String?

    val time: Time?
    fun getAmPm(amPm: Int): String
    fun getImagePath(imageIndex: Int): String
    val calories: Calories?
    val pulse: Pulse?
    val caloriesGraph: CaloriesGraph?
    val stepWidget: Steps?
    //    static ITheme Load(String path) {
    val stepProgress: StepsProgress?
//    };
}